File: app/src/main/java/com/example/contactapp/showContact.java
package com.example.contactapp;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class showContact extends AppCompatActivity {

    String[] ContactInfo;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_show_contact);
        Intent i = getIntent();
        Bundle bundle = i.getExtras();
        ContactInfo = bundle.getStringArray("Contacto");
        TextView Name = (TextView) findViewById(R.id.Name);
        TextView Phone = (TextView) findViewById(R.id.Phone);
        TextView Email = (TextView) findViewById(R.id.Email);
        Name.setText(ContactInfo[0]);
        Phone.setText(ContactInfo[1]);
        Email.setText(ContactInfo[2]);
    }
    public void Regresar (View view){
        Intent i = new Intent(this, Principal.class);
        startActivity(i);
    }
    public void Llamar(){

    }
}

File: app/src/main/java/com/example/contactapp/showContact.kt
package com.example.contactapp

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.TextView
import android.widget.Toast

class showContact : AppCompatActivity() {

    lateinit var global: MyAplication
    internal var ContactInfo: Array<String>? = null
    internal var bundle: Bundle? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_show_contact)
        global = applicationContext as MyAplication
        val i = intent
        bundle = i.extras
        ContactInfo = bundle!!.getStringArray("Contacto")
        val Name = findViewById<View>(R.id.Name) as TextView
        val Phone = findViewById<View>(R.id.Phone) as TextView
        val Email = findViewById<View>(R.id.Email) as TextView
        Name.text = ContactInfo!![0]
        Phone.text = ContactInfo!![1]
        Email.text = ContactInfo!![2]
        Phone.setOnClickListener(object : View.OnClickListener {

            /**
             * Called when a view has been clicked.
             *
             * @param v The view that was clicked.
             */
            override fun onClick(v: View) {
                val Numero = bundle!!.getInt("position")
                Toast.makeText(baseContext, global.misContactos.phone(global.misContactos.Contactos[Numero].telefono), Toast.LENGTH_LONG)
            }
        })
    }

    fun Regresar(view: View) {
        val i = Intent(this, Principal::class.java)
        startActivity(i)
    }
}
